lib/defaultingress/common.go

```diff
@@ -0,0 +1,100 @@
+/*
+  This file is part of Astarte.
+
+  Copyright 2021-23 SECO Mind Srl
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package defaultingress
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/openlyinc/pointy"
+	networkingv1 "k8s.io/api/networking/v1"
+
+	apiv1alpha2 "github.com/astarte-platform/astarte-kubernetes-operator/apis/api/v1alpha2"
+	ingressv1alpha1 "github.com/astarte-platform/astarte-kubernetes-operator/apis/ingress/v1alpha1"
+)
+
+func getCommonIngressAnnotations(cr *ingressv1alpha1.AstarteDefaultIngress, parent *apiv1alpha2.Astarte) map[string]string {
+	apiSslRedirect := pointy.BoolValue(parent.Spec.API.SSL, true) || pointy.BoolValue(cr.Spec.Dashboard.SSL, true)
+	annotations := map[string]string{
+		"nginx.ingress.kubernetes.io/ssl-redirect":   strconv.FormatBool(apiSslRedirect),
```

Comment:
Do we want to base SSL settings for metrics base on API/Dashboard? One might want to expose metrics using plain HTTP on a private subnet while using HTTPS for all other Astarte services

Reply:
The certificate is already available and using HTTPS comes at no additional costs, regardless of the chosen network configuration.